include/kickcat/Gateway.h

```diff
@@ -0,0 +1,40 @@
+#ifndef KICKCAT_GATEWAY_H
+#define KICKCAT_GATEWAY_H
+
+#include "AbstractDiagSocket.h"
+#include "Bus.h"
+#include "Mailbox.h"
+
+namespace kickcat
+{
+    /// \brief Handle external mailbox requests as defined in ETG.8200
+    class Gateway
+    {
+    public:
+        /// \param socket       Socker to communicate with external networks
+        /// \param addMessage   Callback that enable the gateway instance to transfer the request to the EtherCAT bus
+        Gateway(std::shared_ptr<AbstractDiagSocket> socket, std::function<std::shared_ptr<GatewayMessage>(uint8_t const*, int32_t, uint16_t)> addMessage)
+            : socket_{socket}
+            , addMessage_{addMessage}
+            , pendingRequests_{}
+        { }
+        virtual ~Gateway() = default;
+
+        /// \brief   Fetch a request on the external network and process it if any
+        /// \details Try to get a request from the network (non blocking call). If there is data, check the request coherency then
+        ///          transfer it to the network while adding it to the pending requests
```

Comment:
transfer it to the bus instead of the network no ?

Reply:
Right: this is misleading and I am referring to the EtherCAT bus